core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -447,6 +447,16 @@ public Location add(Operation operation) throws IOException {
         }
     }
 
+    /**
+     * The a {@linkplain Location} that will sort after the {@linkplain Location} returned by the last write but before any locations which
+     * can be returned by the next write.
+     */
+    public Location getLastWriteLocation() {
+        try (ReleasableLock lock = readLock.acquire()) {
+            return new Location(current.generation, current.sizeInBytes() - 1, Integer.MAX_VALUE);
```

Comment:
I personally would return `0` - in any case put comments on it why we do it taht way